### PR TITLE
109 get chatroomchat

### PR DESCRIPTION
**Pull request by DLiSWE**

### CODE REVIEW

File: src/controllers/message.controller.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { MessageService } from '@/services';
+import { ObjectId } from 'mongoose';
+import { BaseMessageDto, CreateMessageDto } from '@/dtos';
+
+class MessageController {
+  private messageService = new MessageService();
+
+  public getChatroomMessages = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const id = req.params.chat_id as unknown as ObjectId;
```
Comment (ZohaibManzoor00) [inline src/controllers/message.controller.ts:12]: what does this mean?

File: src/services/message.service.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { BaseChatroomDto, BaseMessageDto, CreateMessageDto } from '@/dtos';
+import { Chatroom, Messages } from '@/entities/nosql';
+import { HttpException } from '@/exceptions';
+import { model, ObjectId } from 'mongoose';
+
+class MessageService {
+  public async getChatroomMessages(id: ObjectId): Promise<BaseMessageDto[]> {
```
Comment (ZohaibManzoor00) [inline src/services/message.service.ts:7]: Can you explain what ObjectId from mongoose does?

Comment (DLiSWE) [reply src/services/message.service.ts:7]: It is an ObjectId typing for typescript for mongodb documents.

File: src/controllers/message.controller.ts (hunk 1 above)
Comment (DLiSWE) [reply src/controllers/message.controller.ts:12]: casting string type chat_id to mongoose's built in ObjectId type, which is mongodb's "primary key".

**Review by ZohaibManzoor00 — DISMISSED** [review]
Looks good

File: src/routes/message.route.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  public readonly path = '/messages';
+  public readonly router = Router();
+  private messageController = new MessageController();
+
+  constructor() {
+    this.initializeRoutes();
+  }
+
+  private initializeRoutes() {
+    this.router.get(`/chatroom/:chat_id`, AuthMiddleware, this.messageController.getChatroomMessages);
```
Comment (Matt-Tan15) [inline src/routes/message.route.ts:16]: I think this should be /chatroom/:chat_id/this.path

Comment (DLiSWE) [reply src/routes/message.route.ts]: Makes sense

File: src/services/message.service.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    const chatroom = model('Chatroom', Chatroom);
+    const findMessages: BaseChatroomDto = await chatroom.findOne({ id });
+    if (!findMessages.messages) throw new HttpException(404, 'No Messages');
```
Comment (raylu511) [inline src/services/message.service.ts:10]: I don't think we really need this. If we get an empty list in the frontend, we just display nothing

File: src/services/message.service.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { BaseChatroomDto, BaseMessageDto, CreateMessageDto } from '@/dtos';
+import { Chatroom, Messages } from '@/entities/nosql';
+import { HttpException } from '@/exceptions';
+import { model, ObjectId } from 'mongoose';
+
+class MessageService {
+  public async getChatroomMessages(id: ObjectId): Promise<BaseMessageDto[]> {
+    const chatroom = model('Chatroom', Chatroom);
+    const findMessages: BaseChatroomDto = await chatroom.findOne({ id });
```
Comment (raylu511) [inline src/services/message.service.ts:9]: nit: ` const { messages }: BaseChatroomDto = await chatroom.findOne({ id }); `

**Review by raylu511 — APPROVED** [review]

**Review by Matt-Tan15 — APPROVED** [review]
you should write a test for it

**Pull request merged**